pkg/morph/deploy/notary.go

```diff
@@ -0,0 +1,872 @@
+					needReset = true
+				}
+			}
+
+			recordExists = true
+		}
+
+		if needReset {
+			prm.logger.Info("calculating signature of the transaction designating Notary role to the committee using local account...")
+
+			sig := prm.localAcc.SignHashable(_actor.GetNetwork(), tx)
+			sig = sharedTxData.unshiftChecksum(sig)
+
+			rec = base64.StdEncoding.EncodeToString(sig)
+
+			l.Info("sending new transaction setting domain record in the NNS...")
+
+			var vub uint32
+			if recordExists {
+				_, vub, err = _actor.SendCall(prm.nnsOnChainAddress, methodNNSSetRecord,
+					domain, int64(nns.TXT), 0, rec)
+			} else {
+				_, vub, err = _actor.SendCall(prm.nnsOnChainAddress, methodNNSAddRecord,
+					domain, int64(nns.TXT), rec)
+			}
+			if err != nil {
+				setDomainRecordTxValidUntilBlock = 0
+				if isErrNotEnoughGAS(err) {
+					prm.logger.Info("not enough GAS to set NNS domain record, will try again later")
+				} else {
+					prm.logger.Error("failed to send transaction setting NNS domain record, will try again later", zap.Error(err))
+				}
+				return
+			}
+
+			setDomainRecordTxValidUntilBlock = vub
+
+			l.Info("transaction setting NNS domain record has been successfully sent, will wait for the outcome")
+
+			return
+		}
+	}, nil
+}
+
+// sharedTransactionData groups transaction parameters that cannot be predicted
+// in a decentralized way and need to be sent out.
+type sharedTransactionData struct {
+	sender          util.Uint160
+	validUntilBlock uint32
+	nonce           uint32
+}
+
+// bytes serializes sharedTransactionData.
+func (x sharedTransactionData) bytes() []byte {
+	b := make([]byte, sharedTransactionDataLen)
+	// fixed size is more convenient for potential format changes in the future
+	copy(b, x.sender.BytesBE())
+	binary.BigEndian.PutUint32(b[util.Uint160Size:], x.validUntilBlock)
+	binary.BigEndian.PutUint32(b[util.Uint160Size+4:], x.nonce)
+	return b
+}
+
+// encodeToString returns serialized sharedTransactionData in base64.
+func (x sharedTransactionData) encodeToString() string {
+	return base64.StdEncoding.EncodeToString(x.bytes())
+}
+
+// decodeString decodes serialized sharedTransactionData from base64.
+func (x *sharedTransactionData) decodeString(s string) (err error) {
+	b, err := base64.StdEncoding.DecodeString(s)
+	if err != nil {
+		return fmt.Errorf("decode shared transaction data from base64: %w", err)
+	}
+
+	if len(b) != sharedTransactionDataLen {
+		return fmt.Errorf("invalid/unsupported length of shared transaction data: expected %d, got %d",
+			sharedTransactionDataLen, len(b))
+	}
+
+	x.sender, err = util.Uint160DecodeBytesBE(b[:util.Uint160Size])
+	if err != nil {
+		return fmt.Errorf("decode sender account binary: %w", err)
+	}
+
+	x.validUntilBlock = binary.BigEndian.Uint32(b[util.Uint160Size:])
+	x.nonce = binary.BigEndian.Uint32(b[util.Uint160Size+4:])
+
+	return nil
+}
+
+const (
+	sharedTransactionDataLen         = util.Uint160Size + 4 + 4
```

Comment:
But then it won't be as convenient doing this (`io.GetVarSize` can help, but it requires implementing `io.Serializable`).